src/h5web/visualizations/line/ErrorBarCurve.tsx

```diff
@@ -23,58 +21,65 @@ function ErrorBarCurve(props: Props): ReactElement {
     ordinates,
     errors,
     color = DEFAULT_COLOR,
-    capLength = CAP_LENGTH,
     visible,
   } = props;
 
   const { abscissaScale, ordinateScale } = useCanvasScales();
 
-  const errorGeometry = useMemo(() => {
-    const points = ordinates.flatMap((val, index) => {
-      const [x, y] = [abscissaScale(abscissas[index]), ordinateScale(val)];
+  const [barsGeometry, capsGeometry] = useMemo(() => {
+    const capPoints: Vector2[] = [];
+    const barSegments: Vector2[] = [];
 
-      if (!Number.isFinite(y)) {
-        return [];
+    for (const index of ordinates.keys()) {
+      const val = ordinates[index];
+      const x = abscissaScale(abscissas[index]);
+      const y = ordinateScale(val);
+      if (!Number.isFinite(x) || !Number.isFinite(y)) {
+        continue;
       }
-
-      const error = errors[index];
+      const error = ordinateScale(errors[index]);
       if (error === 0) {
-        return [];
+        continue;
       }
-      const yerr_min = ordinateScale(val - error);
-      const yerr_max = ordinateScale(val + error);
 
-      return [
-        // First segment: bottom cap
-        ...lineSegment(
-          x - capLength / 2,
-          yerr_min,
-          x + capLength / 2,
-          yerr_min
-        ),
-        // Second segment: bottom error bar
-        ...lineSegment(x, yerr_min, x, y),
-        // Third segment: top error bar
-        ...lineSegment(x, y, x, yerr_max),
-        // Fourth segment: top cap
-        ...lineSegment(
-          x - capLength / 2,
-          yerr_max,
-          x + capLength / 2,
-          yerr_max
-        ),
-      ];
-    });
+      const bottom_cap = abscissaScale(val - error);
+      const top_cap = abscissaScale(val + error);
 
-    const geometry = new BufferGeometry();
-    geometry.setFromPoints(points);
-    return geometry;
-  }, [abscissaScale, abscissas, capLength, errors, ordinateScale, ordinates]);
+      const xyVector = new Vector2(x, y);
```

Comment:
I discovered that we can work with `Vector2` as three.js will automatically convert them in `Vector3` with `z=0`